pkg/discovery/container_builder.go

```diff
@@ -0,0 +1,179 @@
+package discovery
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+	"github.com/turbonomic/turbo-go-sdk/pkg/builder"
+	"github.com/turbonomic/turbo-go-sdk/pkg/supplychain"
+	"github.com/turbonomic/mesosturbo/pkg/data"
+)
+
+// Builder for creating Container Entities to represent the default container Mesos Tasks in Turbo server
+type ContainerEntityBuilder struct {
+	mesosMaster         *data.MesosMaster
+	containerMetricsMap map[string]*data.EntityResourceMetrics
+	builderError        []error
+}
+
+// Build Container EntityDTO using the tasks listed in 'state' json returned from the Mesos Master
+func (cb *ContainerEntityBuilder) BuildEntities() ([]*proto.EntityDTO, error) {
+	glog.V(2).Infof("[BuildEntities] ....")
+	cb.builderError = make([]error, 100, 100)
+	result := []*proto.EntityDTO{}
+	taskMap := cb.mesosMaster.TaskMap
+	for _, task := range taskMap {
+		// skip non running tasks
+		if task.State != "TASK_RUNNING" {
```

Comment:
Shall we merge the logic of creating app and container? Since both of them are parsing from Task.